### PR TITLE
setup routing

### LINKED ISSUE #3: Setup Routing
Go through the user journey. Make notes on which routes are needed. Find out which will be static routes and which ones will be dynamic. We recommend doing this as a small team or a pair. Routes should be simple and informative, e.g.
- /login
- /movies
- /favorites
- /blahblahblah

- [x] setup provider for react-router-dom and all needed config
- [x] setup the layouts with proper nesting
- [x] setup route definitions (be mindful about dynamic routing)
- [x] create empty pages for each route and render "Hello ${route}" on each page

### PR DESCRIPTION
**Pull request by pswu11**
closes #3

### CODE REVIEW

File: src/main.tsx [hunk 1]
```diff
@@ -1,15 +1,81 @@
...
+        path: '/',
+        element: <Home />,
+        index: true,
+      },
+      {
+        path: 'movies',
+        element: <Movies />,
+        children: [
+          {
+            path: ':MovieId',
```
Comment (ReallyBoard123) [inline src/main.tsx:34]: with the current setup, I was not able to reach to cast when I tried movies/1/cast -> was that also true for you?

File: src/pages/Cast.tsx [hunk 2]
```diff
@@ -0,0 +1,7 @@
+// import {useLoaderData} from "react-router-dom"
+// use {Movie} from ../types
+// probably going to need this
+
+export function Cast() {
```
Comment (ReallyBoard123) [inline src/pages/Cast.tsx:5]: optional: calling it CastAndCrew

**Review by ReallyBoard123 — CHANGES_REQUESTED** [review]
suggestion from @dnmct 

There could be multiple ways of achieving this, but one that is probably the cleanest is to think in terms of 

1. NavbarLayout
----- index: Home
----- Movies
----- Bookmarks
----- User

2. MovieLayout (for the dynamic routing) at path: movies/:movieId
----- index: true, MovieDetails
----- Cast at cast 
----- SelectTime
----- SelectSeat
----- Ticket

3. Genres
4. Login

**Comment by ReallyBoard123** [comment]
It was a great attempt. Very tricky problem! Requires shuffling a few things here or there, and then the routing would be set!

File: src/main.tsx (hunk 1 above)
Comment (pswu11) [reply src/main.tsx:34]: Hey, @ReallyBoard123 there's no movies data available, so this movieId is just a placeholder.

**Review by ReallyBoard123 — APPROVED** [review]
🥇

**Pull request merged**